src/libDirectoryService/DirectoryService.cpp

```diff
@@ -677,6 +710,174 @@ bool DirectoryService::ToBlockMessage([[gnu::unused]] unsigned char ins_byte) {
+    return false;
+  }
+
+  uint64_t currentDSEpochNumber =
+      m_mediator.m_dsBlockChain.GetLastBlock().GetHeader().GetBlockNum() + 1;
+  uint64_t loCurrentDSEpochNumber = currentDSEpochNumber - 1;
+  uint64_t hiCurrentDSEpochNumber = currentDSEpochNumber + 1;
+
+  if (!(dsEpochNumber <= hiCurrentDSEpochNumber + 1 &&
+        dsEpochNumber >= loCurrentDSEpochNumber - 1)) {
+    LOG_GENERAL(WARNING,
+                "Update of ds guard network info failure due to not within "
+                "range of expected "
+                "ds epoch loCurrentDSEpochNumber: "
+                    << loCurrentDSEpochNumber
+                    << " hiCurrentDSEpochNumber: " << hiCurrentDSEpochNumber
+                    << " dsEpochNumber: " << dsEpochNumber);
+    return false;
+  }
+
+  // Check timestamp
+  // Allowed only 5 mins. Else consider it obselete.
+  uint64_t allowableTimeForDSNodeNetworkInfoUpdate = 300;
+  if (!VerifyTimestamp(timestamp, allowableTimeForDSNodeNetworkInfoUpdate)) {
+    LOG_GENERAL(WARNING, timestamp << "exceeded "
+                                   << allowableTimeForDSNodeNetworkInfoUpdate
+                                   << " seconds. Current time is "
+                                   << get_time_as_int());
+    return false;
+  }
+
+  {
+    // Update DS committee
+    lock_guard<mutex> g(m_mediator.m_mutexDSCommittee);
+
+    unsigned int indexOfDSGuard;
+    bool foundDSGuardNode = false;
+    for (indexOfDSGuard = 0;
+         indexOfDSGuard < Guard::GetInstance().GetNumOfDSGuard();
+         indexOfDSGuard++) {
+      if (m_mediator.m_DSCommittee->at(indexOfDSGuard).first == dsGuardPubkey) {
+        foundDSGuardNode = true;
+        LOG_GENERAL(INFO,
+                    "[update ds guard] DS guard to be updated is at index "
+                        << indexOfDSGuard << " "
+                        << m_mediator.m_DSCommittee->at(indexOfDSGuard).second
+                        << " -> " << dsGuardNewNetworkInfo);
+        m_mediator.m_DSCommittee->at(indexOfDSGuard).second =
+            dsGuardNewNetworkInfo;
+        break;
+      }
+    }
+
+    // Lookup to store the info
+    if (foundDSGuardNode && LOOKUP_NODE_MODE) {
+      {
+        std::mutex m_mutexLookupStoreForGuardNodeUpdate;
+        DSGuardUpdateStruct dsGuardNodeIden(dsGuardPubkey,
+                                            dsGuardNewNetworkInfo, timestamp);
+        if (m_lookupStoreForGuardNodeUpdate.find(dsEpochNumber) ==
+            m_lookupStoreForGuardNodeUpdate.end()) {
+          vector<DSGuardUpdateStruct> temp = {dsGuardNodeIden};
+          m_lookupStoreForGuardNodeUpdate.emplace(dsEpochNumber, temp);
+          LOG_EPOCH(
+              WARNING, to_string(m_mediator.m_currentEpochNum).c_str(),
+              "[update ds guard] No existing record found for dsEpochNumber "
+                  << dsEpochNumber << ". Adding a new record");
+
+        } else {
+          vector<DSGuardUpdateStruct> temp =
```

Comment:
Probably more efficient to access the map entry for `dsEpochNumber` and doing `emplace_back` directly on the returned vector, instead of creating `temp`.

Reply:
Done!